clio-server/src/main/scala/org/broadinstitute/clio/server/webservice/WgsCramWebService.scala

```diff
@@ -0,0 +1,74 @@
+package org.broadinstitute.clio.server.webservice
+
+import akka.http.scaladsl.server.Directives._
+import akka.http.scaladsl.server._
+import org.broadinstitute.clio.server.service.WgsCramService
+import org.broadinstitute.clio.transfer.model.wgscram.{
+  TransferWgsCramV1Key,
+  TransferWgsCramV1Metadata,
+  TransferWgsCramV1QueryInput
+}
+import org.broadinstitute.clio.util.model.Location
+
+trait WgsCramWebService { self: JsonWebService =>
+
+  def wgsCramService: WgsCramService
+
+  lazy val wgsCramRoutes: Route = {
+    pathPrefix("v1") {
+      pathPrefix("wgscram") {
+        concat(cramGetSchema, cramPostMetadata, cramQuery, cramQueryall)
+      }
+    }
+  }
+
+  private[webservice] val cramPathPrefixKey
+    : Directive1[TransferWgsCramV1Key] = {
+    for {
+      location <- pathPrefix(Location.pathMatcher)
+      project <- pathPrefix(Segment)
+      sampleAlias <- pathPrefix(Segment)
+      version <- pathPrefix(IntNumber)
+    } yield TransferWgsCramV1Key(location, project, sampleAlias, version)
+  }
+
+  private[webservice] val cramPostMetadata: Route = {
+    pathPrefix("metadata") {
+      cramPathPrefixKey { key =>
+        post {
+          entity(as[TransferWgsCramV1Metadata]) { metadata =>
+            complete(wgsCramService.upsertMetadata(key, metadata))
+          }
+        }
+      }
+    }
+  }
+
+  private[webservice] val cramQuery: Route = {
+    path("query") {
+      post {
+        entity(as[TransferWgsCramV1QueryInput]) { input =>
+          complete(wgsCramService.queryMetadata(input))
+        }
+      }
+    }
+  }
+
+  private[webservice] val cramQueryall: Route = {
+    path("queryall") {
```

Comment:
could this be pathPrefix?

Reply:
No, because this should be an exact match on the end of a route (if it was prefix, then "queryallabcdefgh" etc. would match).